src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -105,82 +96,27 @@ public void CopyTo(X509Certificate2Collection collection)
-                certificates = ReadDirectory();
-                Debug.Assert(certificates != null);
-            }
-
-            foreach (X509Certificate2 cert in certificates)
-            {
-                collection.Add(cert);
-            }
-        }
-
-        private List<X509Certificate2> ReadDirectory()
-        {
             if (!Directory.Exists(_storePath))
             {
-                // Don't assign the field here, because we don't have a FileSystemWatcher
-                // yet to tell us that something has been added.
-                return new List<X509Certificate2>(0);
+                return;
             }
 
-            List<X509Certificate2> certs = new List<X509Certificate2>();
-
-            lock (_fileWatcherLock)
+            foreach (string filePath in Directory.EnumerateFiles(_storePath, PfxWildcard))
             {
-                if (_watcher == null)
+                X509Certificate2 cert;
+
+                try
                 {
-                    _watcher = new FileSystemWatcher(_storePath, PfxWildcard)
-                    {
-                        NotifyFilter = NotifyFilters.LastWrite,
-                    };
-
-                    FileSystemEventHandler handler = FlushCache;
-                    _watcher.Changed += handler;
-                    _watcher.Created += handler;
-                    _watcher.Deleted += handler;
-                    // The Renamed event has a different delegate type
-                    _watcher.Renamed += FlushCache;
-                    _watcher.Error += FlushCache;
+                    cert = new X509Certificate2(filePath);
                 }
-
-                // Start watching for change events to know that another instance
-                // has messed with the underlying store.  This keeps us aligned
-                // with the Windows implementation, which opens stores with change
-                // notifications.
-                _watcher.EnableRaisingEvents = true;
-
-                foreach (string filePath in Directory.EnumerateFiles(_storePath, PfxWildcard))
+                catch (CryptographicException)
                 {
-                    X509Certificate2 cert;
-
-                    try
-                    {
-                        cert = new X509Certificate2(filePath);
-                    }
-                    catch (CryptographicException)
-                    {
-                        // The file wasn't a certificate, move on to the next one.
-                        continue;
-                    }
-
-                    certs.Add(cert);
+                    // The file wasn't a certificate, move on to the next one.
+                    continue;
                 }
 
-                // Don't release _fileWatcherLock until _certificates is assigned, otherwise
-                // we may be setting it to a stale value after the change event said to clear it
-                _certificates = certs;
+                collection.Add(cert);
```

Comment:
Seems like you could simplify the loop slightly by making the body of the try:
```C#
collection.Add(new X509Certificate2(filePath));
```
and then getting rid of this line as well as the ```continue;``` in the catch block.